documentation/API-Testing-Guidelines.md

```diff
@@ -1,41 +1,288 @@
-Based on the decision taken in the Commonalities working group and as documented in [API-Readiness-Checklist](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-Readiness-Checklist.md), at least one Gherkin feature file will be added to the main subproject repo and this will fulfill the minimum criteria of readiness with respect to API test cases and documentation. If subprojects also intend to add test implementations, an aligned single implementation that is agreed among all provider implementors could be added to the main subproject repo. If no alignment is possible, each provider implementer will add the test implementation to their own repos.
+## Test plan coverage <a name="coverage"></a>
 
-## Location of feature file <a name="location"></a>
-The feature file will reside under 
-```
-Subproject_Repository/code/Test_definitions/file_name.feature
-for e.g. https://github.com/camaraproject/QualityOnDemand/blob/main/code/Test_definitions/QoD_API_Test.feature
-```
+Based on the decision taken in the Release Management and Commonalities working groups, and as documented in [API-Readiness-Checklist](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-Readiness-Checklist.md), at least one Gherkin feature file will be added to the main subproject repo and this will fulfill the minimum criteria of readiness with respect to API test cases and documentation:
+
+* Release candidates must have at least a basic API test plan, covering sunny day scenarios. Scenarios are expected to validate that the implemented interface is compliant with the specification, in terms of syntax and support for the mandatory aspects. 
+* Any public release must include a full test plan, covering both sunny and rainy day scenarios, to be considered stable. This may include unusual or not explicitly specified error scenarios, semantic validations and corner cases.
+
+If subprojects also intend to add test implementations, an aligned single implementation that is agreed among all provider implementors could be added to the main subproject repo. If no alignment is possible, each provider implementer will add the test implementation to their own repos.
+
+* Test plans are not expected to certify or measure the quality of the implementation in terms of accuracy or veracity of the response, but to validate that the interface is compliant with the API specification and that the API is fully implemented.
+
+* When the API allows options or alternative approaches, the test plan should cover those situations and define the expected behaviour for inputs or options not supported.
+  - Implementations will need to inform in advance to the API tester about which subset of options are supported. 
+
+### Independent operations <a name="independent-operations"></a>
+
+Operations that execute an action or retrieve information from the operator, without modifying a resource, will be tested in its own dedicated feature file. 
```

Comment:
Does this mean we will need two different feature files for QoD Operations ? If yes then what would be significance of it ?

Reply:
QoD is mostly a CRUD API, encapsulating sessions resources, so it lies on the other category. Anyway, it is up to QoD subproject, and the contributors of its test plans, to decide on how many feature files are convenient. Splitting a test plan in several features is a test plan design decision, taking into account the complexity of the test plan. The more complex a test plan is, the most likely is that it will be split into several feature files. As in any software project.

These guidelines are recommendations for the design, so we may make the text less normative: "will be tested in its own dedicated feature file" -> "will be **usually** tested in its own dedicated feature file"

Reply:
@jlurien  OK with us kindly accommodate this in PR file as well

Reply:
Added